compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp

```diff
@@ -0,0 +1,215 @@
+{
+  return ((node->rank() == 1 || node->rank() == 0) && node->size<loco::DataType::FLOAT32>() == 1);
+}
+
+inline void update_with_single_element(luci::CircleConst *fused_node,
+                                       const luci::CircleConst *multiplication)
+{
+  for (uint32_t i = 0; i < fused_node->size<loco::DataType::FLOAT32>(); i++)
+  {
+    fused_node->at<loco::DataType::FLOAT32>(i) *= multiplication->at<loco::DataType::FLOAT32>(0);
+  }
+}
+
+luci::CircleConst *gen_fused_weights(luci::CircleConst *weights,
+                                     const luci::CircleConst *multiplication)
+{
+  auto fused_weights = luci::clone(weights);
+  // Single element multiplication:
+  if (is_single_element(multiplication))
+  {
+    update_with_single_element(fused_weights, multiplication);
+  }
+  // N-size multiplication:
+  else
+  {
+    // Go along channels, multiplication size is ensured to be compatible with channels.
+    auto count = fused_weights->dim(0).value();
+    auto size = fused_weights->dim(fused_weights->rank() - 1).value();
+    float val;
+    for (uint32_t c = 0; c < count; c++)
+    {
+      val = multiplication->at<loco::DataType::FLOAT32>(c);
+      for (uint32_t i = 0; i < size; i++)
+      {
+        fused_weights->at<loco::DataType::FLOAT32>(c * size + i) *= val;
+      }
+    }
+  }
+  return fused_weights;
+}
+
+luci::CircleConst *gen_fused_bias(luci::CircleConst *bias, const luci::CircleConst *multiplication)
+{
+  auto fused_bias = luci::clone(bias);
+  // Single element multiplication:
+  if (is_single_element(multiplication))
+  {
+    update_with_single_element(fused_bias, multiplication);
+  }
+  // N-size multiplication:
+  else
+  {
+    // Go along channels, multiplication size is ensured to be compatible with channels.
+    for (uint32_t i = 0; i < fused_bias->size<loco::DataType::FLOAT32>(); i++)
+    {
+      fused_bias->at<loco::DataType::FLOAT32>(i) *= multiplication->at<loco::DataType::FLOAT32>(i);
+    }
+  }
+  return fused_bias;
+}
+
+/**
+ *  Fuse Mul to FullyConnected if the multiplied value is a channel(last dimension)-wise constant
+ *
+ *  BEFORE
+ *                |
+ *      [CircleFullyConnected]
+ *                |
+ *           [CircleMul]
+ *                |
+ *
+ *  AFTER
+ *                |
+ *       [CircleFullyConnected]   [CircleMul] (dead)
+ *                |
+ *
+ */
+bool fuse_mul_with_fc(luci::CircleMul *mul)
+{
+  // Sanity check:
+  RETURN_FALSE_UNLESS(mul);
+  // Allow Mul node only with FLOAT32 data type:
+  RETURN_FALSE_UNLESS(mul->dtype() == loco::DataType::FLOAT32);
+  // Check if any FC node connects to Mul.
+  // Find the pattern of Mul(FC, CircleConst):
+  luci::CircleFullyConnected *fc = nullptr;
+  luci::CircleConst *multiplication = nullptr;
+  RETURN_FALSE_UNLESS(luci::fill(&fc, &multiplication).with_commutative_args_of(mul));
+  // Make sure that FullyConnected has only one successor:
+  RETURN_FALSE_UNLESS(loco::succs(fc).size() == 1);
```

Comment:
Most of our passes don't check this.
Is this really necessary?
Please add a comment for the reason.

Reply:
It's to avoid previously discussed scenario, ref: https://github.com/Samsung/ONE/pull/13607#discussion_r1715072728

I have added comment with details.